frontend/app/routes/_gcweb-app.update-info.tsx

```diff
@@ -48,21 +48,24 @@ export async function action({ request }: ActionFunctionArgs) {
     phoneNumber: z.string().refine(isPhoneNumber, { message: 'Invalid phone number' }),
   });
 
-  const formData = Object.fromEntries(await request.formData()) as UserInfo;
-  const validFormData = formDataSchema.safeParse(formData);
+  const formData = Object.fromEntries(await request.formData());
+  const parseResult = formDataSchema.safeParse(formData);
 
-  if (!validFormData.success) {
+  if (!parseResult.success) {
     return json({
-      errors: validFormData.error.flatten(),
+      errors: parseResult.error.flatten(),
       formData: formData as Partial<z.infer<typeof formDataSchema>>,
     });
   }
 
-  await updateUserInfo('1234', formData);
+  const userInfo = parseResult.data;
+  await updateUserInfo(await getUserId(), userInfo);
+
   return redirect('/update-info-success');
 }
 
 export async function loader() {
```

Comment:
```suggestion
export async function loader({ request }: LoaderFunctionArgs) {
```

Reply:
@sebastien-comeau any reason to add the `request` even though it isn't being used?

Reply:
@gregory-j-baker suggested to keep it there for faster debug purposes in my latest [PR](https://github.com/DTS-STN/canadian-dental-care-plan/pull/80#discussion_r1455478612).

Reply:
It's useful when debugging and troubleshooting to see the incoming request (headers, URL params, etc).

Reply:
[Fixed](https://github.com/DTS-STN/canadian-dental-care-plan/pull/82/files#diff-2bec7e9aba6317cf91db5121aa046967350d305eba20a20d32dfa45459862afbR31)